### ISSUE TITLE
cgo cannot parse _cgo_.o as ELF, Mach-O, PE or XCOFF and go tool dist FAILED go_bootstrap install std cmd exit status 1 

### ISSUE THREAD

**Issue by Azaretdodo** [golang/go#62236]
### What version of Go are you using (`go version`)?

<pre>
$ go version 1.20.7

</pre>

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env windows amd64 

</pre></details>

### What did you do?

I was try to install go from the src folder from go paste on windows system32 with the script all.bat but i happen these two lines of errors below

```
cgo: cannot parse $WORK\b251\_cgo_.o as ELF, Mach-O, PE or XCOFF
go tool dist : FAILED: C:\Windows\System32\go1.20.7\pkg\tool\windows_amd64\go_bootstrap install std cmd: exit status 1

```

### What did you expect to see?

The success of the installing of go with all.bat script 

### What did you see instead?

```
cgo: cannot parse $WORK\b251\_cgo_.o as ELF, Mach-O, PE or XCOFF
go tool dist : FAILED: C:\Windows\System32\go1.20.7\pkg\tool\windows_amd64\go_bootstrap install std cmd: exit status 1

```

**Comment by bcmills** [comment]
Sounds like a problem with your C compiler or linker.

Please post the complete output of `go env` as requested in the issue template, and also include information about the C toolchain (`gcc --version` or similar).

**Comment by Azaretdodo** [comment]
hello dear worker in IT,


maybe the problem when by the setup because `go env` answer a problem : 

```
PS C:\Windows\System32\go1.20.7\src> go env
go: cannot find GOROOT directory: C:\Windows\System32\go1.20.7\bin\go.exe
```

so i am on the system windows 64 byte and use only msys for compile thus i paste below the variable use by go and the c compiler : 

![image](https://github.com/golang/go/assets/30596513/6b3e229b-bb8b-4ea6-ba13-a1d9379e89b6)

thanks you in advance for your help,

Regards.


Azaretdodo.

**Comment by bcmills** [comment]
Yeah, that's not going to work. `GOROOT` and `GOPATH` do not mean what you seem to think they mean. You should not need to set either of those explicitly, no matter where you install the Go toolchain.

I suggest unsetting both of those variables and reading through https://go.dev/doc/code to familiarize yourself with the Go development workflow.

**Comment by Azaretdodo** [comment]
you are just bad because goroot is always in windows => system32 for windows and gopath is always in the directory user for both system after for the documentation go it is always bugged because sometimes it is helping finaly i think you answer aside the meaning of the problem,

thanks you in advance to really help myself,

Regards.


Azaretdodo.

**Comment by bcmills** [comment]
I'm sorry, I can't really parse that reply. 😅

Unlike many projects, the Go project does not use GitHub Issues for general discussion or asking questions. GitHub Issues are used for tracking bugs and [proposals](https://go.dev/s/proposal-process) only.

For asking questions, see:

[Stack Overflow](https://stackoverflow.com/questions/tagged/go) with questions tagged "go" or "golang"

[The Go Forum](https://forum.golangbridge.org/), a web-based forum

[Gophers Discord](https://discord.gg/golang) "Get live support and talk with other gophers on the Go Discord"

[Gophers Slack](https://gophers.slack.com/), use the [invite app](https://invite.slack.golangbridge.org/) for access. The #general channel is a good starting point.

[Go Community on Hashnode](https://hashnode.com/n/go) with questions and posts tagged with "go"

[The golang-nuts mailing list](https://groups.google.com/d/forum/golang-nuts)

[Subreddit for Go](https://www.reddit.com/r/golang/)

IRC channel #go-nuts on Libera
